unitTests/ui/ResultListTest.ts

```diff
@@ -353,6 +353,13 @@ export function ResultListTest() {
       expect($$(test.cmp.element).hasClass('coveo-hidden')).toBe(false);
     });
 
+    it('enableScrollBackTop should be set', () => {
+      test = Mock.optionsComponentSetup<ResultList, IResultListOptions>(ResultList, {
+        enableScrollToTop: false
+      });
+      expect(test.cmp.options.enableScrollToTop).toBe(false);
+    });
+
```

Comment:
I think we should also have a test that checks when `true`, the logic in `scrollBackToTop` is called, and when false, it is not.

Reply:
Any easy way to do it? 
Since everything in the ResultList is private it would take a lot of work to do implement so I thought we could create a functional test for this feature

Reply:
Yes, looking at the implementation, if you set the `infiniteScrollContainer` option, it looks like calling `scrollToTopIfEnabled` should either set scrollTop to 0 if `infiniteScrollContainer` is an element, or call the scrollTo method if it is an instance of `window`.

Reply:
Ok so I am having a problem creating the test and I hope I can get some help.

I am trying to create the test with the infiniteScrollContainer option set.  I’m trying
to set `scrollTop` of the element to something other than 0  to be able to check after the query that it is to 0.

I can’t set `scrollTop` to anything. I try having a with different height for my htmlElement, with and without the overflow-y attribute. I even created a lot of fake results to have result in the element and i still can't set it.

Heres my code for now: 


      it('to do later', done => {
        const infiniteScrollContainer = $$('div');
        infiniteScrollContainer.setAttribute('style', 'height: 100px; overflow-y: scroll;');

        const option: IResultListOptions = {
          enableInfiniteScroll: true,
          infiniteScrollContainer: infiniteScrollContainer.el,
          enableScrollToTop : true
        };

        const promiseResults = new Promise(resolve => {
          resolve(FakeResults.createFakeResults(50));
        });

        test = Mock.basicComponentSetup<ResultList>(ResultList, option);
        test.cmp.displayMoreResults(50);
        promiseResults.then(() => {
          infiniteScrollContainer.el.scrollTop = 500;
          expect(infiniteScrollContainer.el.scrollTop).toBe(500)
          Simulate.query(test.env);
          expect(infiniteScrollContainer.el.scrollTop).toBe(0)
        })
      });

Any help will be helpful

Reply:
It looks like `scrollTop` is a setter with restrictions on the values it will accept if the element is not scrollable: https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollTop

I think the simplest solution would be to use a mock element as opposed to a real one. Does this work?

```
const mockEl = {
    scrollTop: <non-zero value>
} as HtmlElement;

const option: IResultListOptions = {
    ...,
    infiniteScrollContainer: mockEl,
    ...
};
```

Reply:
yes thank you